File: PG2Lab8/PG2Lab8/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PG2Lab8
{
    class Employee
    {
        protected string Name;
        protected string LastName;
        protected int Age;
        protected int Salary;
        public Employee(string n = "John", string l = "doe", int a = 1, int s = 10)
        {
            Name = n;
            LastName = l;
            Age = a;
            Salary = s;
        }
        public virtual string Info()
        {
            string str = "";
            str += Name + " ";
            str += LastName + " ";
            str += Age + " ";
            str += Salary;

            return str;
        }
    }
}

File: McElroy_Sabian_AbstractBase/McElroy_Sabian_AbstractBase/Employee.cpp
#include "stdafx.h"
#include "Employee.h"


Employee::Employee(char * n, char * d, int s)
{
	name = n;
	department = d;
	salary = s;
}


Employee::~Employee()
{

}

void Employee::DisplayRecord() {
	cout << "Name: " << name << "\nDepartment: " << department << "Salary: " << salary;
}